### DISCUSSION TITLE
Getting logs from two Docker containers

### DISCUSSION THREAD

**Discussion by iavor-k** [logdyhq/logdy-core#58] [Q&A] [answered]
Hello,
I've setup logdy as a service listening on ports 8551, 8552.. 8555. Docker is running multiple containers and I want to redirect each log to a different port, so I can distinguish them on the logdy ui. According to the offical website docs, I should to use:
```
$ docker logs my-container1 --follow --tail=10 | logdy forward 8551
$ docker logs my-container2 --follow --tail=10 | logdy forward 8552
```
But when I enter the first line, the program shows :
`INFO[14:42:15.882] Accept stdin and forward to port              port=8551`
It doesn't exit and keeps the bash busy. I can't enter the second line without interrupting the logging of "my-container1".
Is there some way to deamonise the commands. I tried with setsid but it doesn't seem to work.
My OS is ubuntu 24.04

**Comment by PeterOsinski** [comment, marked as answer]
The reason you can't enter multiple log forwarding commands in the same terminal is because `docker logs --follow` is a "blocking" operation - it continuously streams logs in real-time and keeps the process running. This is by design, as it needs to stay active to monitor and forward new log entries as they appear.

When you run:
```bash
docker logs my-container1 --follow --tail=10 | logdy forward 8551
```

The terminal stays "busy" because:
1. The `--follow` flag tells Docker to continuously watch for new logs
2. The pipe (`|`) streams these logs to logdy in real-time
3. logdy needs to keep running to receive and forward these logs to its specified port

This is why you need separate terminals/tabs - each streaming operation needs its own process to run continuously. Opening new tabs allows you to run multiple independent streaming processes simultaneously.

The simplest approach is to use Ubuntu's native terminal tabs. Just press `Ctrl+Shift+T` to open a new tab for each container you want to monitor. Then run your logdy forwarding commands in each tab:

```bash
# First tab
docker logs my-container1 --follow --tail=10 | logdy forward 8551

# Switch to new tab (Ctrl+Shift+T)
docker logs my-container2 --follow --tail=10 | logdy forward 8552
```

You can switch between tabs using `Alt+1`, `Alt+2`, etc. or `Ctrl+PgUp`/`Ctrl+PgDown`.

No additional software or complex configuration needed - it's all built into your Ubuntu terminal.
